torch/distributions/relaxed_categorical.py

```diff
@@ -0,0 +1,110 @@
+import torch
+from torch.autograd import Variable
+from torch.nn.functional import log_softmax
+from torch.distributions import constraints
+from torch.distributions.utils import clamp_probs, broadcast_all
+from torch.distributions.categorical import Categorical
+from torch.distributions.distribution import Distribution
+from torch.distributions.transformed_distribution import TransformedDistribution
+from torch.distributions.transforms import ExpTransform
+
+
+def _log_softmax(x, *args, **kwargs):
+    if isinstance(x, torch.Tensor):
```

Comment:
1. Since `Tensor` and `Variable` are being merged, this might not be necessary.
2. Prefer `torch.is_tensor()`, since `torch.Tensor` is an alias rather than a base class:
```py
isinstance(torch.FloatTensor([0]), torch.Tensor)
# True
isinstance(torch.DoubleTensor([0]), torch.Tensor)
# False
torch.Tensor is torch.FloatTensor
# True
```

Reply:
1.  I'm hoping it's not necessary soon, but it definitely is right now because `log_softmax` takes `Variables` and raises when called on a `Tensor`. Can we leave the shim until the merge? 
2. Definitely, didn't realize this was the case.